### PR TITLE
[JENKINS-34285] Script and signature approval tests.

### PR DESCRIPTION
**Pull request by andresrc**
See [JENKINS-34285](https://issues.jenkins-ci.org/browse/JENKINS-34285)

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/test/acceptance/plugins/groovy_postbuild/GroovyPostBuildStep.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import org.jenkinsci.test.acceptance.po.Control;
+import org.jenkinsci.test.acceptance.po.Describable;
+import org.jenkinsci.test.acceptance.po.Job;
+import org.jenkinsci.test.acceptance.po.PostBuildStep;
+
+/**
+ * Groovy post build step. Use {@code form-element-path} plugin.
+ * Assumes "path" form element availability.
+ * Paths are usually available when using the default Jenkins controller.
+ * (Available thanks to pre-installed form-element-path plugin.)
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/groovy_postbuild/GroovyPostBuildStep.java]: All steps does. I do not think that the testsuite is of much of a vale without the plugin.

File: src/test/java/plugins/ScriptSecurityPluginTest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    private void shouldSucceed(FreeStyleJob job) {
+        login(USER);
+        job.scheduleBuild().shouldSucceed();
+    }
+
+    @Test
+    public void scriptNeedsApproval() throws Exception {
+        final FreeStyleJob job = createFailedJob("def a = 4", false);
```
Comment (rsandell) [inline src/test/java/plugins/ScriptSecurityPluginTest.java]: why would this fail??

Comment (andresrc) [inline src/test/java/plugins/ScriptSecurityPluginTest.java]: because it does not use the sandbox, so the script requires approval.

**Comment by rsandell** [comment]
:bee:

File: src/main/java/org/jenkinsci/test/acceptance/plugins/script_security/PendingSignature.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+public class PendingSignature extends CapybaraPortingLayerImpl {
+    private final WebElement block;
+
+    PendingSignature(Injector injector, WebElement block) {
+        super(injector);
+        this.block = block;
+    }
+
+    private void click(String buttonName) {
+        block.findElement(by.button(buttonName)).click();
```
Comment (jglick) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/script_security/PendingSignature.java]: What if there are multiple approvals pending? This API will not work.

Comment (jglick) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/script_security/PendingSignature.java]: I see, `findSignature` does that lookup.

**Comment by jglick** [comment]
retro:bee:

**Pull request merged**